app/tutorials/example-applications/src/main/java/org/eclipse/mosaic/app/tutorial/emergencybrake/EmergencyBrakeApp.java

```diff
@@ -0,0 +1,220 @@
+        float curDeceleration = (speedDifference / timeDifference) / 9.81f;
+        if (curDeceleration < -(configuration.emergencyBrakeThresh)) {
+            brakeCounter++;
+        } else {
+            brakeCounter = 0;
+        }
+
+        // 300: threshold in ms, braking time must be longer
+        if (brakeCounter == (configuration.minimalBrakeDuration / configuration.minimalTimerCallInterval)) {
+            getLog().info("Detected emergency brake in progress");
+
+            // Vehicle info
+            GeoPoint vehicleLongLat = getOs().getPosition();
+            String roadId = getOs().getNavigationModule().getRoadPosition().getConnection().getId();
+
+            // Prepare the DENMessage
+            MessageRouting routing =
+                    getOs().getAdHocModule().createMessageRouting().viaChannel(AdHocChannel.SCH1).topoBroadCast();
+            Denm denm = new Denm(routing, new DenmContent(getOs().getSimulationTime(), vehicleLongLat, roadId,
+                    SensorType.SPEED, 1, curSpeed, curDeceleration * 9.81f, null,
+                    null, null));
+            getLog().debug("Sender position: " + denm.getSenderPosition());
+            getLog().debug("Sender speed at event time: " + denm.getCausedSpeed() + "m/s");
+            getLog().debug("Sender deceleration at event time: " + denm.getSenderDeceleration() + "m/s2");
+            getLog().debug("RoadId on which the event take place: " + denm.getEventRoadId());
+            getLog().debug("DENMessage successfully filled");
+
+            // Send the DENMessage
+            getOs().getAdHocModule().sendV2xMessage(denm);
+            getLog().info("Sent DENMessage");
+        }
+
+        // Update time and speed buffer for calculation of deceleration
+        lastTime = getOs().getSimulationTime();
+        lastSpeed = curSpeed;
+
+        getLog().debug("t=" + getOs().getSimulationTime()
+                + "\t SpeedDifference: " + speedDifference
+                + "\t timeDifference: " + timeDifference
+                + "\t curDeceleration: " + curDeceleration);
+    }
+
+    @Override
+    public void onShutdown() {
+    }
+
+    @Override
+    public void onMessageReceived(ReceivedV2xMessage receivedV2xMessage) {
+
+    }
+
+    @Override
+    public void onAcknowledgementReceived(ReceivedAcknowledgement acknowledgement) {
+
+    }
+
+    @Override
+    public void onCamBuilding(CamBuilder camBuilder) {
+
+    }
+
+    @Override
+    public void onMessageTransmitted(V2xMessageTransmission v2xMessageTransmission) {
+
+    }
+
+    /**
+     * Checks for an obstacle detected by the vehicle sensors.
+     * If an obstacle is detected an emergency brake will be performed.
+     *
+     * @param previousVehicleData the previous state of the vehicle
+     * @param updatedVehicleData  the updated state of the vehicle
+     */
+    @Override
+    public void onVehicleUpdated(@Nullable VehicleData previousVehicleData, @Nonnull VehicleData updatedVehicleData) {
```

Comment:
this method is quite important (and public), and should be located further up in this file